modules/apps/27-interchain-accounts/controller/keeper/migrations.go

```diff
@@ -55,6 +55,11 @@ func (m Migrator) AssertChannelCapabilityMigrations(ctx sdk.Context) error {
 // MigrateParams migrates the controller submodule's parameters from the x/params to self store.
 func (m Migrator) MigrateParams(ctx sdk.Context) error {
 	if m.keeper != nil {
+		// set KeyTable if it has not already been set
+		if !m.keeper.legacySubspace.HasKeyTable() {
```

Comment:
@damiannolan do we know if this is necessary? I was trying to pickup on why we have this. It seems to me the key table doesn't set the values. When I modify the tests, if the key table isn't set, this will still panic as the param set would need to be set as well

I feel like it is safe to assume that this is run during an in-place migration which means that any chain which has run InitGenesis on a binary which did not use the param set in the keeper store, would have already initialized and set the key table in the param subspace. This should be done on a per chain execution basis, not per node?

Reply:
Or maybe the issue is that the binary would be restarted, so the key table gets wiped and our tests have a hard time mocking this behaviour because they don't have the pre-set kv store values?

Reply:
yea, the codecov doesn't seem to be hitting the if statement

Reply:
Looks like the best way to test this is to set the param keys in the kv store instead of initializing the key table. I think it's just:
```go
store := prefix.NewStore(ctx.KVStore(app.GetKey(paramstypes.StoreKey)), append(icacontrollertypes.SubModuleName, '/'))
store.Set(param.Key, param.Value)
```